imblearn/under_sampling/_prototype_selection/_instance_hardness_threshold.py

```diff
@@ -126,6 +126,9 @@ def _validate_estimator(self):
                 isinstance(self.estimator, ClassifierMixin) and
                 hasattr(self.estimator, 'predict_proba')):
             self.estimator_ = clone(self.estimator)
+            self.estimator_.set_params(random_state=self.random_state)
```

Comment:
We should use [this](https://github.com/scikit-learn/scikit-learn/blob/813b601e9e212251e9ac1e3d5f85fdcb9aa80457/sklearn/ensemble/base.py#L22) function to set the `random_state` of the estimator. (I know that this function is intended for internal use. We could vendor it).  Apart from that, I believe you should get the `random_state` object using the `check_random_state` utility function in `_fit_resample` and pass it here and later in `cross_val_predict`